pylint/checkers/variables.py

```diff
@@ -1042,6 +1082,12 @@ def _undefined_and_used_before_checker(
             if action is VariableVisitConsumerAction.CONTINUE:
                 continue
             if action is VariableVisitConsumerAction.CONSUME:
+                assert found_nodes is not None, "Cannot consume an empty list of nodes."
```

Comment:
Is this necessary? As in, doesn't the code break anyway `if not isinstance(found_nodes, list)` because of L1090? This assert doesn't seem to "add" much. Or is this because of `mypy` complaining?
If the latter, could you add a `TODO` which says to remove this after this has been typed correctly? If `_check_consumer` is typed this shouldn't be a problem anymore.

Reply:
Yes, it's the latter, mypy complains that if `found_nodes` is None, then the `+=` operation will fail. I'll add a TODO.